source-map.d.ts

```diff
@@ -43,56 +63,267 @@ export interface MappingItem {
     name: string;
 }
 
-export class SourceMapConsumer {
-    static GENERATED_ORDER: number;
-    static ORIGINAL_ORDER: number;
+export interface Mapping {
+    generated: Position;
+    original: Position;
+    source: string;
+    name?: string;
+}
 
-    static GREATEST_LOWER_BOUND: number;
-    static LEAST_UPPER_BOUND: number;
+export interface CodeWithSourceMap {
+    code: string;
+    map: SourceMapGenerator;
+}
 
-    constructor(rawSourceMap: RawSourceMap);
+export interface SourceMapConsumer {
+    /**
+     * Compute the last column for each generated mapping. The last column is
```

Comment:
Is it necessary to copy in the documentation here?  I ask because it seems likely to me that future updates to the main copy will probably not update this copy; which would be bad.

Reply:
The documentation is what will appear when hovering over names. It's not ideal, but it's the only way TypeScript will pick it up. Virtually all of this module is copied from the current typings (obtained via `npm install @types/source-map`).